### PR TITLE
IBX-1465: Prioritize current site access when choosing language

### PR DESCRIPTION
**Pull request by mateuszdebinski**
| Question      | Answer
| ------------- | ---
| Tickets       | [IBX-1465](https://issues.ibexa.co/browse/IBX-1465)
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | no
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

#### Checklist:
- [x] Coding standards (`$ composer fix-cs`)
- [x] Ready for Code Review

### CODE REVIEW

File: src/lib/UI/Config/Provider/Languages.php [hunk 1]
```diff
@@ -34,10 +38,12 @@ class Languages implements ProviderInterface
     public function __construct(
         LanguageService $languageService,
         ConfigResolverInterface $configResolver,
+        SiteAccess $siteAccess,
         array $siteAccesses
     ) {
         $this->languageService = $languageService;
         $this->configResolver = $configResolver;
+        $this->siteAccess = $siteAccess;
```
Comment (ciastektk) [inline src/lib/UI/Config/Provider/Languages.php:46]: Current SiteAccess shouldn't be taken from `SiteAccessService::getCurrent`?

Comment (mateuszdebinski) [reply src/lib/UI/Config/Provider/Languages.php:46]: in most cases, from what I checked the code, we use `eZ\Publish\Core\MVC\Symfony\SiteAccess`, but if you think that we should use `SiteAccessService`, then I don't see a problem to change to `SiteAccessService`

`SiteAccessService::getCurrent` i can't find i v2.5

Comment (mateuszdebinski) [reply src/lib/UI/Config/Provider/Languages.php:46]: In v2.5 there is no `SiteAccessService::getCurrent()` but it will be used when creating PR for v3

Comment (adamwojs) [reply src/lib/UI/Config/Provider/Languages.php:46]: AFAIR SiteAccessService::getCurrent is available since v3.x

**Review by konradoboza — APPROVED** [review]

**Review by mikadamczyk — APPROVED** [review]

**Review by bogusez — APPROVED** [review]

**Pull request merged**